crd-generator/collector/src/main/java/io/fabric8/crd/generator/collector/JandexIndexer.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.generator.collector;
+
+import org.jboss.jandex.ClassSummary;
+import org.jboss.jandex.Index;
+import org.jboss.jandex.Indexer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Stream;
+
+/**
+ * Allows to create a Jandex index based on given class files, directories and JAR archives.
+ * <p>
+ * Unlike {@link Index#of(File...)}, this implementation performs a recursive scan
+ * of the directories and enforces some limits.
+ * </p>
+ */
+class JandexIndexer {
+
+  private static final Logger log = LoggerFactory.getLogger(JandexIndexer.class);
+
+  private static final String CLASS_FILE_SUFFIX = ".class";
+  private static final String JAR_FILE_SUFFIX = ".jar";
+
+  private int maxJarEntries = 100000;
+  private long maxBytesReadFromJar = 100000000; // 100 MB
+  private long maxClassFileSize = 1000000; // 1 MB
```

Comment:
Do we need limits?

Reply:
These seem reasonable limits.
We can keep it this way.
If users complain, then we can make this configurable in a future release.

Reply:
Ok, let's keep it.

Just as a note:
I'm a little bit unsure if those checks make sense overall...
I mean, if a project cannot trust it's dependencies, then the project has more problems than those from a zip bomb. But if we think towards the CRD-Generator CLI, it's some kind of own input validation and kind of useful.